python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1523,6 +1540,25 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
                 maskNameList=list(self.config.maskListToInterpolate)
             )
 
+        # An extra config validation logic that checks for compatibility
+        # among the config options for amp offset correction.
+        if self.config.doApplyAmpOffset and not self.config.doMeasureAmpOffset:
+            raise RuntimeError("doApplyAmpOffset requires doMeasureAmpOffset to be True.")
+
+        # Measure amp offset corrections within the CCD.
+        if self.config.doMeasureAmpOffset:
```

Comment:
And this would just be `if self.config.doAmpOffset: self.ampOffset.run(ccdExposure)`.